Cleanup cards

iantrich opened the pull request.

iantrich committed:
Cleanup cards

thomasloven commented:
I pulled this and every card seems to work, at least...

iantrich committed:
Update hui-iframe-card.ts

balloob commented on src/panels/lovelace/cards/hui-markdown-card.ts:
```diff
@@ -64,35 +61,39 @@ export class HuiMarkdownCard extends LitElement implements LovelaceCard {
+  static get styles(): CSSResult {
+    return css`
+      :host {
+        @apply --paper-font-body1;
```
We can remove this. This has never worked.

balloob replied:
`@apply` is a deprecated spec that Polymer supports, but will never become a standard.

balloob commented on src/panels/lovelace/cards/hui-picture-entity-card.ts:
```diff
@@ -132,37 +129,39 @@ class HuiPictureEntityCard extends LitElement implements LovelaceCard {
+      }
+
+      .footer {
+        @apply --paper-font-common-nowrap;
```
This won't work.

balloob commented on src/panels/lovelace/cards/hui-picture-glance-card.ts:
```diff
@@ -177,44 +176,47 @@ class HuiPictureGlanceCard extends LitElement implements LovelaceCard {
+      }
+
+      .box {
+        @apply --paper-font-common-nowrap;
```
This won't work.

balloob commented on src/panels/lovelace/cards/hui-entities-card.ts:
```diff
@@ -128,38 +127,42 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
+      }
+
+      .header {
+        @apply --paper-font-headline;
```
Won't work.

balloob commented on src/panels/lovelace/cards/hui-entities-card.ts:
```diff
@@ -128,38 +127,42 @@ class HuiEntitiesCard extends LitElement implements LovelaceCard {
+      }
+
+      .header .name {
+        @apply --paper-font-common-nowrap;
```
Won't work

iantrich committed:
address review comments

iantrich commented:
Going to rename all `Config` interfaces to match card name before merging

thomasloven commented:
You can't just drop the `@apply` lines.

You have to manually add everything they did, e.g. `font-size: 24px;` for `.header`
![skarmavbild 2019-03-01 kl 23 45 40](https://user-images.githubusercontent.com/1299821/53670793-3c7dcf00-3c7c-11e9-967c-181238e970b6.png)

iantrich commented:
oh, derp

iantrich committed:
Apply paper-styles

https://github.com/PolymerElements/paper-styles/blob/v3.0.1/classes/typography.js

I didn't find `paper-font-common-nowrap` in anything after v1.0.0 but did end up applying what I found here: https://chromium.googlesource.com/chromium/src/+/923ede74ebc57633583795a84b33e994f7dcd4fa/third_party/polymer/v1_0/components/paper-styles/typography.html

iantrich committed:
Added comments on paper-style usage

iantrich commented:
paper-styles should be copied over now

balloob commented:
🎉 !!
